### ISSUE TITLE
plot_method="plotly" - ignores k_col, k_row

### ISSUE THREAD

**Issue by talgalili** [talgalili/heatmaply#48]
Example:

```r
library(heatmaply)
heatmaply(mtcars, 
          k_col = 2, k_row = 4,
          plot_method="plotly")

```

![image](https://cloud.githubusercontent.com/assets/976006/24837620/d49a5cb6-1d40-11e7-8e64-3593a35f42e0.png)

**Comment by Alanocallaghan** [comment]
@talgalili I've addressed this with one semi-major shortcoming. Unfortunately, the colors are not necessarily in the same order as in the standard ggplot plots. I think this is an issue with either how the colors come back from `dendextend::get_leaves_branches_col` or how plotly assigns colors to the line segments. Not sure how to fix this, it's certainly an improvement.

eg: ://imgur.com/a/XinTw

**Comment by Alanocallaghan** [comment]
To be clear I don't think it's a bug with either, just a shortcoming in how they work in this situation

**Comment by talgalili** [comment]
It is indeed an improvement :)

I would prefer them to be consistent (since some users may supply Rowv/Colv, in which case the colors might be meaningful to the user).
If you supply Rowv, would it still give different results?

For example:
```r
library(heatmaply)
library(dendextend)
dend <- mtcars %>% dist %>% hclust %>% as.dendrogram %>% color_branches(4)
heatmaply(mtcars, Rowv = dend)
heatmaply(mtcars, Rowv = dend, plot_method = "plotly")

```

(BTW, I noticed you used k_row = 1, this would need to be managed later on in haetmapr, once you push your commits  I can fix it).

**Comment by Alanocallaghan** [comment]
It was actually a simple fix. Since ggplot sorts everything it takes in, you just need to sort the colors for plotly, too.

`k_row = 1` is mostly a workaround to make all branches black, which avoids duplicating the plotting method later on. This means it should work identically for user-supplied or heatmaply-generated dendrograms, since heatmapr now overwrites any coloring present I believe.

By the way, I had until now disabled hoverinfo for the plotly dendrograms until I could figure out some sensible way to do things, should I continue with that or do you have any strong feelings on what should be done there?

**Comment by talgalili** [comment]
Glad you fixed it.
Regarding the hover info for the dendrogram, I believe it should behave like in the regular one, which is to show the height of the nearby branch (it can be informative in some cases).

As for ggplot, I'm not sure I understand how the sorting is done. If you can share an example, that may be helpful for me to understand (or just tell me which line of code solved it). Thanks :)

**Comment by Alanocallaghan** [comment]
Makes sense; I tried to include the labels as well but unfortunately you get a mix of numeric and character values.

For character input, ggplot generally uses the equivalent of `levels(factor(input))` on all of the character inputs it takes. This will be the equivalent of `sort(input)` unless it is already a factor with levels. The line I changed is line 256 in `plots.R`, just to sort the colors before applying them to the plot.

EG, to change the order of facets/colors/etc, you change either the way the values are ordered, or change the levels of the factor:

    library(ggplot2)
    ggplot(diamonds, aes(x=carat, y=depth)) + geom_point() + facet_wrap(~cut)
    diamonds$cut <- factor(diamonds$cut, levels = rev(levels(diamonds$cut)))
    ggplot(diamonds, aes(x=carat, y=depth)) + geom_point() + facet_wrap(~cut)
